src/V3PerPackage/PerBatchContext.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.V3PerPackage
+{
+    public class PerBatchContext
+    {
+        public PerBatchContext(PerWorkerContext workerContext, string name)
```

Comment:
A hierarchy of contexts is used which mirrors the hierarchy of work and allows data to be shared at the right scope.

"global" is data shared across multiple processes (or multiple machines).
"process" is data shared across multiple `Task`s (a.k.a. workers) in one process.
"worker" is data specific to a single worker. A worker processes one batch at a time (serially).
"batch" is data specific to a set of packages given to V3. This is essentially a catalog commit.
"package" is data specific to a single package.